app/src/main/java/com/example/muisicsvice/service/MusicService.kt

```diff
@@ -16,9 +16,175 @@ import android.util.Log
 import androidx.core.app.NotificationCompat
 import com.example.muisicsvice.ui.MainActivity
 import com.example.muisicsvice.R
-import com.example.muisicsvice.model.Song
+import com.example.muisicsvice.data.model.Song
 import java.util.*
 
-class MusicService {
+class MusicService : Service(), OnPreparedListener,
+    MediaPlayer.OnErrorListener,
+    OnCompletionListener {
+    //media player
+    private var player: MediaPlayer? = null
+    //song list
+    private var songs: MutableList<Song>? = null
+    //current position
+    private var songPosn = 0
+    //binder
+    private val musicBind: IBinder = MusicBinder()
+    //title of current song
+    private var songTitle = ""
+    override fun onCreate() {
+        //create the service
+        super.onCreate()
+        //initialize position
+        songPosn = 0
+        //random
+        //create player
+        player = MediaPlayer()
+        //initialize
+        initMusicPlayer()
+    }
 
+    fun initMusicPlayer() {
+        //set player properties
+        player!!.setWakeMode(
+            applicationContext,
+            PowerManager.PARTIAL_WAKE_LOCK
+        )
+        player!!.setAudioStreamType(AudioManager.STREAM_MUSIC)
+        //set listeners
+        player!!.setOnPreparedListener(this)
+        player!!.setOnCompletionListener(this)
+        player!!.setOnErrorListener(this)
+    }
+
+    //pass song list
+    fun setList(theSongs: MutableList<Song>?) {
+        songs = theSongs
+    }
+
+    //binder
+    inner class MusicBinder : Binder() {
+        val service: MusicService
+            get() = this@MusicService
+    }
+
+    //activity will bind to service
+    override fun onBind(intent: Intent): IBinder? {
+        return musicBind
+    }
```

Comment:
Chuyển sang dùng `expression`
```suggestion
    override fun onBind(intent: Intent) = musicBind
```